app/services/chunkers/chunk_record.rb

```diff
@@ -3,16 +3,27 @@ module Chunkers
   # embedding. If a chunk record is created without a separately specific `embedding_content`,
   # the `content`` is used instead.
   class ChunkRecord
-    attr_reader :content
+    attr_reader :excerpt, :surrounding_content, :headings, :location_summary
 
-    def initialize(content:, embedding_content: nil)
-      @content = content
-      @embedding_content = embedding_content
+    def initialize(excerpt:, surrounding_content: nil, headings: nil, location_summary: nil)
+      @excerpt = excerpt
+      @surrounding_content = surrounding_content || excerpt
+      @headings = headings
+      @location_summary = location_summary
     end
 
     def embedding_content
-      # Use content as the key for embedding if no separate key specified
-      @embedding_content || content
+      # Assemble the embedding content used to generate the
+      # vector embeddings
+      text = ""
+      if headings
+        text << headings << "\n"
+      end
+      if location_summary
+        text << location_summary << "\n"
+      end
+      # Use document excerpt if no surrounding content provided (compatibility)
+      text << (surrounding_content || excerpt) << "\n"
```

Comment:
Most of the time Archyve is using two newlines to separate semantically different blocks, the same way one would separate paragraphs in a text or markdown doc.

I'm not sure how much it helps the LLM, but: I think it's better to be consistent, it reads better to a user, and one would think it matches training data better.

Reply:
Yes, makes sense. The expectation should be that the fields themselves aren't required to (nor should they) include terminating newlines, and then the aggregator will use two new lines between them.